package.json

```diff
@@ -13,6 +13,7 @@
     "@azure-tools/typespec-providerhub": "0.31.0",
     "@azure-tools/typespec-client-generator-core": "0.31.0",
     "@typespec/compiler": "0.45.0",
+    "@typespec/versioning": "0.45.0",
```

Comment:
If hte goal is to pin all the intermediate deps we should probably also pin 
- `@typespec/http`
- `@typespec/rest`
- `@typespec/openapi`

Reply:
I was debating whether to pin all intermediate deps or only the ones currently causing a problem.  I think the only package we **need** to pin now is `versioning` due to the range peer dep from `typespec-apiview`, because `npm` likes to default to `latest` if it matches any requested range.

But it probably wouldn't hurt to pin all `@typespec` and `@azure-tools` packages we are using to avoid any suprises.

@timotheeguerin, @weshaggard: Any opinions?  I can create another PR to pin all packages if you want to compare them.

Reply:
I'd say it is better to be consistent here and pin them all. It isn't a big cost as they are all using the same version.
But I also not going to feel too strongly about either.

Reply:
Also pin `@typespec/lint`?

Reply:
It looks like we already pin all the packages we use under `@azure-tools`.

Reply:
> Also pin `@typespec/lint`

this one I say maybe no because we should move to the built in linter when I get that done and that package will be obsolete